File: solutions/lua/sum-of-multiples/1/sum-of-multiples.lua
local function sum(numbers)
  local result_t = {}
  local result = 0
  return {
    to = function(limit)
      for _, i in pairs(numbers) do
        local ii = i
        while ii < limit do
          table.insert(result_t, ii)
          ii = ii + i
        end
      end
      table.sort(result_t);
      for i, ii in pairs(result_t) do
        local _, yy = next(result_t, i)
        if ii == yy then
          ii = 0
        end
        result = result + ii
      end
      return result
    end
  }
end

return sum

File: solutions/go/sum-of-multiples/1/sum_of_multiples.go
package summultiples

func SumMultiples(limit int, divisors ...int) int {
	list := map[int]int{}
	for _, i := range divisors {
		if limit < i || i == 0 {
			break
		}
		for ii := 1; ii < limit; ii++ {
			if ii%i == 0 {
				list[ii]++
			}
		}
	}
	sum := 0
	for i := range list {
		sum += i
	}

	return sum
}
